### PR TITLE
Added support for arrays

### PR DESCRIPTION
**Pull request by codepitbull**
**Motivation**

Resolves #28937

**Changes**
OPCUA was not able to use arrays for Southbound.

### CODE REVIEW

File: modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java [hunk 1]
```diff
@@ -130,6 +135,49 @@ public BuiltinJsonSchema() {
...
+            propertiesNode.set("type", new TextNode(ARRAY_DATA_TYPE));
+
+            //0 for a dimension means unlimited
+            if(maxSize > 0) {
+                propertiesNode.set("maxItems", new LongNode(maxSize));
+            }
+            final ObjectNode itemsNode = objectMapper.createObjectNode();
+            propertiesNode.set("items", itemsNode);
+
+            if (dimensions.length == 1) { //we are the last element
```
Comment (DC2-DanielKrueger) [inline modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java:172]: guess the comment was kind of duplicated

File: modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java [hunk 2]
```diff
@@ -130,6 +135,49 @@ public BuiltinJsonSchema() {
...
+        requiredAttributes.add("value");
+        rootNode.set("required", requiredAttributes);
+        return rootNode;
+    }
+
+    public static void populatePropertiesForArray(final @NotNull ObjectNode propertiesNode,
+                                                  final @NotNull BuiltinDataType builtinDataType,
+                                                  final @NotNull ObjectMapper objectMapper,
+                                                  final @NotNull UInteger[] dimensions) {
+            final long maxSize = dimensions[0].longValue();
```
Comment (DC2-DanielKrueger) [inline modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java:161]: safety check on the dimensions.lemgth would be nice.

**Review by DC2-DanielKrueger — APPROVED** [review]

File: modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java (hunk 2 above)
Comment (codepitbull) [reply modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java:161]: Happens in JsonSchemaGenerator before this code is invoked.

Comment (DC2-DanielKrueger) [reply modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java:161]: I know but the method should assert this. It is not part of the method contract.

Comment (codepitbull) [reply modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/mqtt2opcua/BuiltinJsonSchema.java:165]: Added a check

**Review by DC2-DanielKrueger — APPROVED** [review]

**Pull request merged**